### DISCUSSION TITLE
Is there a way to send webhook before the action is done? Sync webhooks

### DISCUSSION THREAD

**Discussion by mmkhitaryan** [directus/directus#10586] [Feature Requests]
When I delete an object the webhook is sent with a request body like this:
```
{"event":"items.delete","accountability":{"user":"86396cd7-958d-44da-9ead-b140b90f8e99","role":"feaf9c77-dd21-4bda-b179-1cea590c5b51"},"payload":[15710],"collection":"user"}
```
But the problem is that the webhook is sent after the data is deleted from the database. And when I look up for a user with id 15710 on my database there is no because it has already been deleted. I think it would be cool if directus waited for the response from the webhook, then actually deleted the data only after receiving a response from the webhook.

**Comment by rijkvanzanten** [comment]
Webhooks are intended to be used as an asynchronous response to an event that happened. If you want to block the deletion based on an external check, you could use a regular filter hook 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
